### PR TITLE
remove app dependency on email lib

### PR DESCRIPTION
**Pull request by asood123**

### CODE REVIEW

File: server/lib/email.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ loadTemplates();
 /**
  * Mailgun wrapper
  */
-const EmailLib = (app) => {
+const EmailLib = () => {
```
Comment (sedubois) [inline server/lib/email.js]: shouldn't this line disappear entirely?

File: test/lib.email.test.js [hunk 2]
```diff
@@ -1,11 +1,55 @@
...
+      return nm;
+    });
+    done();
+  });
+
+  // stub the transport
+  beforeEach(done => {
+    sendMailStub = sinon.stub(nm, 'sendMail', (object) => {
+      console.log("sendMail: ", object);
+      return Promise.resolve; // this is the line that doesn't seem to return properly
```
Comment (arnaudbenard) [inline test/lib.email.test.js]: @asood123 `return Promise.resolve();`

Comment (asood123) [inline test/lib.email.test.js]: Tried that. It says

```
 1) Root Suite lib/email sends the thankyou.fr email template:
     TypeError: expecting a function but got [object Object]
      at Function.Promise.promisify (node_modules/bluebird/js/release/promisify.js:270:15)
      at sendMessage (email.js:53:20)
      at Object.generateFromTemplateAndSend [as send] (email.js:116:10)
      at Context.<anonymous> (lib.email.test.js:73:21)
      at Context.<anonymous> (lib.email.test.js:41:5)
      at Context.<anonymous> (lib.email.test.js:32:5)
```

Comment (arnaudbenard) [inline test/lib.email.test.js]: @asood123 Then it's the promisification that went wrong, can you try using 
```js
new Promise((resolve, reject) =>
 // insert sendMessage here
)
```

Comment (asood123) [inline test/lib.email.test.js]: That did it. Thanks @arnaudbenard. Painful tutorial of stubs and mocks.

Comment (arnaudbenard) [inline test/lib.email.test.js]: Working though the pain 🌵

**Pull request merged**